### ISSUE TITLE
Graph display incorrect Mode from Radio Button under graph window

### ISSUE THREAD

**Issue by mboland** [ArduPilot/apm_planner#1006]
With the Radio button under the graph window all modes are displayed as "Stabilise by Radio" regardless of the actual mode.

![screen shot 2017-02-13 at 12 19 28 pm](https://cloud.githubusercontent.com/assets/3221133/22871081/4cdfc940-f1f9-11e6-8469-f4473f8b0c6f.png)

When the Mode/Mode log entry is selected the Mode is displayed correctly.

![screen shot 2017-02-13 at 12 18 57 pm](https://cloud.githubusercontent.com/assets/3221133/22871101/7735a11a-f1f9-11e6-822a-d67a378d3cc0.png)

![screen shot 2017-02-13 at 12 19 43 pm](https://cloud.githubusercontent.com/assets/3221133/22871105/83bf8964-f1f9-11e6-8525-a7ecd22d39a2.png)

Checked in 2.0.23 - 2.0.24 - 2.0.24Daily (apm_planner_2.0.24-38-g6aeba43_osx)

Addition: I have been opening the text files created by Mission Planner. When I just now opened a .bin file directly in apm_planner_2.0.24-38-g6aeba43_osx it displayed the Modes correctly from the Radio Button under the graph window but the log value Mode and ModeNum display the same numeric values.

**Comment by Arne-W** [comment]
I will check that...

**Comment by Arne-W** [comment]
All right - I can not reproduce this issue. @mboland could you please upload the logfile which causes the issue?
Did I understand you right - you opened a .bin file with the Mission Planner, exported the log with Mission Planner to a .tlog file and than opened this exported tlog with the APM Planner which leads to this misbehaviour of the Mode arrows? Opening the bin file directly with the APM Planner does not show this issue?
I have the suspicion that there is something fishy with the Mission Planner export.

**Comment by mboland** [comment]
I have mission Planner on a WinLapTop in the workshop which I use to download the logs.
Running latest beta version and it converts the .bin to .log on download.
I use MacOS in the office for analysis and initial setups.
APM Planner has the best graphing facilities.

So, I will usually just select the .log file to open in APMPlanner and I get what you see.
Using the .bin file APMPlanner opens  much quicker but if Mode/Mode is selected a numeric graph is produced rather than labels.
![screen shot 2017-02-14 at 6 45 15 am](https://cloud.githubusercontent.com/assets/3221133/22902641/71b74eae-f281-11e6-9399-ca1a0e95b84f.png)
Guess I will just use the .bin files from now on.

**Comment by Arne-W** [comment]
Ah okay - looks like a Mission Planner export issue.
As we have reimplemented the the log parsing and the internal data storage we changed the behaviour of the labeling. The labels are only supported for the 4 message types which have a checkbox below the graph. At the moment we support MODE, ERROR, EVENT and MESSAGE events to have labels. All data that can be selected through the tree view on the right will produce a graph but no labels anymore.
I thought it could be sometimes useful to have both representations. Only measurements which contain a string as value can not be plotted as a graph - they only have the label-arrow representation.
